src/window.rs

```diff
@@ -132,6 +114,60 @@ impl AppWindow {
             self.maximize();
         }
     }
+
+    fn build_headerbar(&self) {
+        let mode_switch = granite::ModeSwitch::builder()
+            .primary_icon_name("display-brightness-symbolic")
+            .secondary_icon_name("weather-clear-night-symbolic")
+            .primary_icon_tooltip_text(gettext("Light Background"))
+            .secondary_icon_tooltip_text(gettext("Dark Background"))
+            .valign(gtk::Align::Center)
+            .build();
+
+        let gtk_settings = gtk::Settings::default().expect("Unable to get GtkSettings object");
+        mode_switch
+            .bind_property("active", &gtk_settings, "gtk-application-prefer-dark-theme")
+            .bidirectional()
+            .build();
+
+        let header_bar = gtk::HeaderBar::builder().show_title_buttons(true).build();
+
+        header_bar.style_context().add_class("default-decoration");
```

Comment:
Use add_css_class, this method is deprecated

Reply:
Nice!